### PR TITLE
SC68402: Add app manager RBAC requirements 

### PR DESCRIPTION
**Pull request by paigecalvert**
Enterprise user docs:
- New section on RBAC installation requirements: https://deploy-preview-934--replicated-docs.netlify.app/enterprise/installing-general-requirements#rbac-requirements
- Updates prerequisites in the Installing in Existing Cluster topic: https://deploy-preview-934--replicated-docs.netlify.app/enterprise/installing-existing-cluster#prerequisites

Reference docs:
- Added a couple new cross-refs to description of minimal RBAC options in the Application CR topic: https://deploy-preview-934--replicated-docs.netlify.app/reference/custom-resource-application#requireminimalrbacprivileges
- Add some of the missing flags in the `upgrade` command topic: https://deploy-preview-934--replicated-docs.netlify.app/reference/kots-cli-admin-console-upgrade
- Update the description of the relevant flags in the `install` command topic: https://deploy-preview-934--replicated-docs.netlify.app/reference/kots-cli-install

Vendor docs:
- Rewrote the vendor topic about app manager RBAC: https://deploy-preview-934--replicated-docs.netlify.app/vendor/packaging-rbac

Story: https://app.shortcut.com/replicated/story/68402/app-manager-specific-rbac-requirements

### CODE REVIEW

File: docs/enterprise/installing-general-requirements.md [hunk 1]
```diff
@@ -64,15 +59,178 @@ To install the admin console on an existing cluster, the cluster must meet the f
...
+
+* **Minimum App manager RBAC permissions**: In some cases, it is not possible to grant the user `* * *` permissions in the target namespace. For example, an organization might have security policies that prevent this level of permissions.
+
+  If the user installing or upgrading the app manager cannot be granted `* * *` permissions in the namespace, then they can instead request the minimum RBAC permissions required by the app manager. Using the minimum app manager RBAC permissions also requires manually creating a ServiceAccount, Role, and RoleBinding for the app manager, rather than allowing the app manager to automatically create a Role with `* * *` permissions.
+
+  To use the minimum app manager RBAC permissions to install or upgrade:
+
+   1. Ensure that the user has the minimum RBAC permissions required by the app manager. The following Role lists the minimum RBAC permissions under `rules`:
+
+    ```yaml
```
Comment (paigecalvert) [inline docs/enterprise/installing-general-requirements.md:111]: Do you think that showing the YAML for this Role is the best way to present this information? Or might people prefer being able to see a list in plain text about the permissions they need?

File: docs/enterprise/installing-general-requirements.md [hunk 2]
```diff
@@ -64,15 +59,178 @@ To install the admin console on an existing cluster, the cluster must meet the f
...
+      - apiGroups: [""]
+        resources: ["pods/log", "pods/exec"]
+        verbs: ["get", "list", "watch", "create"]
+      - apiGroups: ["batch"]
+        resources: ["jobs/status"]
+        verbs: ["get", "list", "watch"]
+      ```
+    :::note
+    The minimum RBAC requirements can vary slightly depending on the cluster's Kubernetes distribution and the version of the app manager.
+    :::
```
Comment (paigecalvert) [inline docs/enterprise/installing-general-requirements.md:150]: > The minimum RBAC requirements can vary slightly depending on the cluster's Kubernetes distribution and the version of the app manager.

Is there a recommended action that the user should take based on this information? "reach out to your vendor if you get a permissions error"?

File: docs/vendor/packaging-rbac.md [hunk 3]
```diff
@@ -1,33 +1,12 @@
-# Configuring Role-Based Access Control
+# Configuring App Manager RBAC
```
Comment (paigecalvert) [inline docs/vendor/packaging-rbac.md:1]: ^ wanted to clarify that this was talking about RBAC for the app manager, and not for the vendor portal (or something else)

File: docs/vendor/packaging-rbac.md [hunk 4]
```diff
@@ -79,68 +66,40 @@ subjects:
...
 
-The RBAC permissions are set during the initial installation. The admin console runs using the assumed identity and cannot change its own authorization. Changing the RBAC scope from cluster to namespace or from namespace to cluster affects only new installations. Existing installations continue to run with their current authorization.
-For applications that need to elevate their permission from namespace to cluster, we recommend including a preflight check to ensure the permission is available.
+   To expand the scope of RBAC for the app manager from namespace-scoped to cluster-scoped, Replicated recommends that you include a preflight check to ensure the permission is available in the cluster.
```
Comment (paigecalvert) [inline docs/vendor/packaging-rbac.md:105]: > To expand the scope of RBAC for the app manager from namespace-scoped to cluster-scoped, Replicated recommends that you include a preflight check to ensure the permission is available in the cluster.

I was a little confused by this suggestion to add a preflight check. It feels like it does not align with the content above, which says that you can't change the kots RBAC permissions for existing installs anyway

File: docs/vendor/packaging-rbac.md [hunk 5]
```diff
@@ -79,68 +66,40 @@ subjects:
...
 
-It is possible to use namespace-scoped access for Operators and multi-namespace applications.
-During the installation, if there are `additionalNamespaces` specified in the Application manifest, Roles and RoleBindings are created to give the admin console access to all specified namespaces.
+* **Velero namespace access**: Namespace-scoped RBAC does not grant access to the namespace where Velero is installed in the cluster. Velero is a requirement for configuring backup and restore with snapshots. 
 
-To enable namespace-scoped access for an application:
+   To set up snapshots when the app manager has namespace-scoped access, users can run the `kubectl kots velero ensure-permissions` command. This command creates additional Roles and RoleBindings to allow the necessary cross-namespace access. For more information, see [`velero ensure-permissions`](/reference/kots-cli-velero-ensure-permissions/) in the kots CLI documentation.
```
Comment (paigecalvert) [inline docs/vendor/packaging-rbac.md:93]: Do you think that we need to add this as a requirement in the Enterprise User docs? Maybe here under "Wildcard Permissions"?

Screen Shot 2023-02-08 at 3 30 26 PM

File: docs/enterprise/installing-general-requirements.md (hunk 1 above)
Comment (cbodonnell) [reply docs/enterprise/installing-general-requirements.md:111]: I see some value in showing the YAML as it allows the user to copy-paste, and potentially adjust to their needs.

Comment (cbodonnell) [reply docs/enterprise/installing-general-requirements.md:111]: I see that we provide the full YAML for everything just below, so maybe a table could be useful here if it's more readable.

File: docs/enterprise/installing-general-requirements.md (hunk 2 above)
Comment (cbodonnell) [reply docs/enterprise/installing-general-requirements.md:150]: Good question.  Some users may be knowledgeable enough to figure it out on their own, but others would definitely need assistance if they hit this roadblock, so it seems reasonable.  Do we direct users to reach out to their vendor in any other sections of the docs?

File: docs/vendor/packaging-rbac.md (hunk 4 above)
Comment (cbodonnell) [reply docs/vendor/packaging-rbac.md:105]: I can see how that might mean:

"To expand the scope of RBAC for the app manager from namespace-scoped to cluster-scoped **for new installations**, Replicated recommends..."

Though it's not exactly clear from how it's worded.

File: docs/reference/kots-cli-admin-console-upgrade.md [hunk 6]
```diff
@@ -13,11 +13,13 @@ This command supports all [global flags](kots-cli-global-flags) and also:
...
+| `--ensure-rbac`             | bool   | When `false`, KOTS does not attempt to create the RBAC resources necessary to manage applications. **Default:** `true`. If a role spec is needed, use the [generate-manifests](kots-cli-admin-console-generate-manifests) command. |
+| `--skip-rbac-check`         | bool   | Set to `true` to bypass RBAC check. **Default:** `false` |
+| `--strict-security-context` | bool   | Set to explicitly enable strict security contexts for all kots pods and containers. This may not work for some storage providers. **Default**: false    |
+| `--use-minimal-rbac`        | bool   |  When set to `true`, KOTS RBAC permissions are limited to the namespace where it is installed. To use `--use-minimal-rbac`, the application must support namespace-scoped installations and the user must have the minimum RBAC permissions required by KOTS in the target namespace. For a complete list of requirements, see [Namespace-scoped RBAC Requirements​](/enterprise/installing-general-requirements#namespace-scoped) in _Installation Requirements_. **Default**: `false` |
```
Comment (cbodonnell) [inline docs/reference/kots-cli-admin-console-upgrade.md:22]: There are a handful of registry-related flags missing from this reference page, though that's not really in the scope of this PR:

```
      --kotsadm-namespace string   set to override the namespace of kotsadm images. used for airgapped installations.
      --kotsadm-registry string    set to override the registry of kotsadm images. used for airgapped installations.
      --registry-password string   password to use to authenticate with the application registry. used for airgapped installations.
      --registry-username string   username to use to authenticate with the application registry. used for airgapped installations.
```

They are described in the `kots install` [reference](https://docs.replicated.com/reference/kots-cli-install).

File: docs/vendor/packaging-rbac.md (hunk 5 above)
Comment (cbodonnell) [reply docs/vendor/packaging-rbac.md:93]: We have this page: https://docs.replicated.com/enterprise/snapshots-velero-installing-config, which we link to from here: https://docs.replicated.com/enterprise/snapshots-config-workflow#configuring-backup-storage-for-existing-clusters.  Might not be a bad idea to link to it from other RBAC-related areas of the enterprise docs.

File: docs/enterprise/installing-general-requirements.md (hunk 2 above)
Comment (paigecalvert) [reply docs/enterprise/installing-general-requirements.md:150]: Definitely--yes that works. 👍

File: docs/reference/kots-cli-admin-console-upgrade.md (hunk 6 above)
Comment (paigecalvert) [reply docs/reference/kots-cli-admin-console-upgrade.md:22]: Yeah, noticed that too. The kots CLI docs could probably use a more thorough update. I made this follow up story: https://app.shortcut.com/replicated/story/69060/kots-cli-docs-missing-flags

File: docs/reference/kots-cli-admin-console-upgrade.md [hunk 7]
```diff
@@ -13,11 +13,13 @@ This command supports all [global flags](kots-cli-global-flags) and also:
...
-| `--ensure-rbac`             | bool   | When set, KOTS will skip RBAC configuration at upgrade time. (default false) if a role spec is needed, use the [generate-manifests](kots-cli-admin-console-generate-manifests) command.                                                      |
-| `--strict-security-context` | bool   | Set to explicitly enable strict security contexts for all kots pods and containers. This may not work for some storage providers. Default: false                                                                                                  |
+| `--with-minio`              | bool   | When set, KOTS will deploy a local minio instance for storage and attempt to change any minio-based snapshots (hostpath and NFS) to the [local-volume-provider](https://github.com/replicatedhq/local-volume-provider) plugin **Default:** `true` |
+| `--ensure-rbac`             | bool   | When `false`, KOTS does not attempt to create the RBAC resources necessary to manage applications. **Default:** `true`. If a role spec is needed, use the [generate-manifests](kots-cli-admin-console-generate-manifests) command. |
```
Comment (jonquil2002) [inline docs/reference/kots-cli-admin-console-upgrade.md:19]: s/ role spec /  role specification

File: docs/reference/kots-cli-admin-console-upgrade.md [hunk 8]
```diff
@@ -13,11 +13,13 @@ This command supports all [global flags](kots-cli-global-flags) and also:
...
+| `--skip-rbac-check`         | bool   | Set to `true` to bypass RBAC check. **Default:** `false` |
+| `--strict-security-context` | bool   | Set to explicitly enable strict security contexts for all kots pods and containers. This may not work for some storage providers. **Default**: false    |
```
Comment (jonquil2002) [inline docs/reference/kots-cli-admin-console-upgrade.md:21]: s/ pods / Pods

File: docs/vendor/packaging-rbac.md (hunk 3 above)
Comment (jonquil2002) [reply docs/vendor/packaging-rbac.md:1]: Yay!

**Review by jonquil2002 — DISMISSED** [review]
lgtm

**Pull request merged**